server/subscribe.go

```diff
@@ -377,6 +387,45 @@ func (p *Plugin) addSubscriptionTemplate(instanceID types.ID, newSubscriptionTem
 	})
 }
 
+func (p *Plugin) editSubscriptionTemplate(instanceID types.ID, modifiedSubscriptionTemplate *SubscriptionTemplate, client Client) error {
+	subKey := keyWithInstanceID(instanceID, templateKey)
+	return p.atomicModify(subKey, func(initialBytes []byte) ([]byte, error) {
+		subscriptionTemplates, err := SubscriptionTemplatesFromJSON(initialBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		oldSubscriptionTemplate, ok := subscriptionTemplates.Templates.ByID[modifiedSubscriptionTemplate.ID]
+		if !ok {
+			return nil, errors.New("existing subscription template does not exist")
+		}
+
+		oldProjectKey := ""
+		if oldSubscriptionTemplate.Filters.Projects.Len() == 1 {
+			oldProjectKey = oldSubscriptionTemplate.Filters.Projects.Elems()[0]
+		}
+
+		newProjectKey := ""
+		if modifiedSubscriptionTemplate.Filters.Projects.Len() == 1 {
+			newProjectKey = modifiedSubscriptionTemplate.Filters.Projects.Elems()[0]
+		}
+
+		if err = p.validateSubscriptionTemplate(modifiedSubscriptionTemplate, instanceID, client, newProjectKey); err != nil {
+			return nil, err
+		}
+
+		subscriptionTemplates.Templates.delete(oldProjectKey, oldSubscriptionTemplate.ID)
+		subscriptionTemplates.Templates.add(modifiedSubscriptionTemplate, newProjectKey)
```

Comment:
It would be great if the `projectKey` is the first parameter in the `add` function.